### PR TITLE
docs: add a API stability document

### PR DESCRIPTION
**Pull request by ansiwen**
Lets work out an API stability document with this PR.

Signed-off-by: Sven Anderson <sven@redhat.com>

### CODE REVIEW

File: docs/api-stability.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+deprecated APIs will not be removed, however they are still deprecated and only
+in maintanence mode. We usually don't make improvements for these APIs and we
+can't guarantee optimal performance.
+
+## Preview
+
+This is a level for APIs that are fresh and might need further refinements in
+following releases. These are marked with _PREVIEW_ in the documentation and are
+only available, if the build tag `ceph_preview` is set. There might be breaking
+changes in future releases reagarding preview APIs. Usually new exported APIs
```
Comment (phlogistonjohn) [inline docs/api-stability.md:32]: Spelling: `regarding`

**Review by phlogistonjohn — COMMENTED** [review]
Awesome start. I'll do a 2nd more thoughtful pass soon but this looks like a really solid foundation to me. One minor typo/spelling mistake highlighted below.

File: docs/api-stability.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+in maintanence mode. We usually don't make improvements for these APIs and we
+can't guarantee optimal performance.
+
+## Preview
+
+This is a level for APIs that are fresh and might need further refinements in
+following releases. These are marked with _PREVIEW_ in the documentation and are
+only available, if the build tag `ceph_preview` is set. There might be breaking
+changes in future releases regarding preview APIs. Usually new exported APIs
+are introduced with this level first and become stable with a later release.
```
Comment (phlogistonjohn) [inline docs/api-stability.md:33]: Let's be a tad bit more specific wrt the stabilization schedule. Something like:
> Usually new exported APIs are introduced with this level first and become stable within two future releases unless a major change to the API is required which then "resets" the counter. The schedule for preview APIs becoming stable is tracked in a separate document (link TDB).

**Review by phlogistonjohn — CHANGES_REQUESTED** [review]
I have one content change suggested. We don't need to take my exact wording but I do think we need to say something along these lines or it's too easy to forget to stabilize apis.

**Review by phlogistonjohn — DISMISSED** [review]
Newest revision looks good. I am approving to show my approval of the content, but do please squash the changes prior to removing the label, thanks!

**Comment by ansiwen** [comment]
> do please squash the changes prior to removing the label, thanks!

Sure, that's what I set the label for. 😉

**Review by phlogistonjohn — APPROVED** [review]
LGTM, thanks!

**Pull request merged**